apps/coscrad-frontend/src/components/resources/factories/thumbnail-resource-detail-container-factory.tsx

```diff
@@ -0,0 +1,38 @@
+import { ResourceType } from '@coscrad/api-interfaces';
+import { FunctionalComponent } from '../../../utils/types/functional-component';
+import { BibliographicReferenceDetailPresenter } from '../bibliographic-references/bibliographic-reference-detail.presenter';
+import { BookDetailPresenter } from '../books/book-detail.presenter';
+import { MediaItemDetailPresenter } from '../media-items/media-item-detail.presenter';
+import { PhotographDetailPresenter } from '../photographs/photograph-detail.presenter';
+import { SongDetailPresenter } from '../songs/song-detail.presenter';
+import { SpatialFeatureDetailPresenter } from '../spatial-features/spatial-feature-detail.presenter';
+import { TermDetailPresenter } from '../terms/term-detail.presenter';
+import { TranscribedAudioDetailPresenter } from '../transcribed-audio/transcribed-audio-detail.presenter';
+import { VocabularyListDetailPresenter } from '../vocabulary-lists/vocabulary-list-detail.presenter';
+
+// TODO Define thumbnail specific presenters
+const lookupTable: { [K in ResourceType]: FunctionalComponent } = {
```

Comment:
This is a bit outside the box, but I wonder if we could use annotation here via a "Decorator" (that we wrap statically instead of letting the compiler do it) along with Reflection.

Reply:
`Decorate` utility could wrap the Presenter but return a result with the same API so that the decoration is transparent to the  compiler.